feat: 프로젝트별 concurrency 설정 지원

글로벌 concurrency 외에 프로젝트별 concurrency 오버라이드 지원. config 3곳 수정 필요 (types, defaults, validator). 관련: src/queue/job-queue.ts, src/types/config.ts. 워크플랜 M1-2

## AI Quartermaster - 파이프라인 실패

자동 구현에 실패했습니다.

**에러**: Cannot find module '/home/by/.ai-quartermaster/src/pipeline/checkpoint.js' imported from /home/by/.ai-quartermaster/src/cli.ts

수동 확인이 필요합니다.

## AI Quartermaster - 파이프라인 실패

자동 구현에 실패했습니다.

**에러**: Cannot find module '/home/by/.ai-quartermaster/src/pipeline/checkpoint.js' imported from /home/by/.ai-quartermaster/src/cli.ts

수동 확인이 필요합니다.

--- a/tests/queue/job-store.test.ts
+++ b/tests/queue/job-store.test.ts
@@ -9,7 +9,7 @@ describe("JobStore", () => {
   let store: JobStore;
 
   beforeEach(() => {
-    dataDir = join(tmpdir(), `aq-jobstore-test-${Date.now()}`);
+    dataDir = join(tmpdir(), `aq-jobstore-test-${Date.now()}-${Math.random().toString(36).slice(2)}`);
     mkdirSync(dataDir, { recursive: true });
     store = new JobStore(dataDir);
   });

